lib/facil/fio.c

```diff
@@ -307,7 +307,7 @@ static inline int fio_clear_fd(intptr_t fd, uint8_t is_open) {
   if (fio_data->max_protocol_fd < fd) {
     fio_data->max_protocol_fd = fd;
   } else {
-    while (fio_data->max_protocol_fd &&
+    while (fio_data->max_protocol_fd > fd &&
```

Comment:
Initially I though "Wow! how didn't I see it"... but then I looked again and I think the reading here makes the logic a bit less obvious...

The code assumes that `fio_data->max_protocol_fd > fd` but the code doesn't know if `fd_data(fd).open` is true. `fd` might be now closed and actually higher than what `max_protocol_fd` should be.

This loop stops at either `max_protocol_fd == 0` or `fd_data(fio_data->max_protocol_fd).open` , so there's no risk of an underflow.

However there might be a bug in the if statement above.